examples/end2end_tree_reconstruction_test.py

```diff
@@ -0,0 +1,356 @@
+    )
+
+    plt.style.use("dark_background")
+    fig, axes = plt.subplots(3 if fossils else 2, 2)
+
+    frames["true"] = alifestd_collapse_unifurcations(frames["true"])
+    true_df_no_lengths = alifestd_mark_node_depth_asexual(frames["true"])
+
+    reconst_df_no_lengths = frames["reconst"].copy()
+    frames["true"]["origin_time"] = frames["true"]["depth"]
+    frames["reconst"]["origin_time"] = frames["reconst"]["hstrat_rank"]
+    true_df_no_lengths["origin_time"] = true_df_no_lengths["node_depth"]
+
+    if fossils:
+        frames["true_dropped_fossils"] = alifestd_collapse_unifurcations(
+            frames["true_dropped_fossils"]
+        )
+        frames["true_dropped_fossils"]["origin_time"] = frames[
+            "true_dropped_fossils"
+        ]["depth"]
+        frames["reconst_dropped_fossils"]["origin_time"] = frames[
+            "reconst_dropped_fossils"
+        ]["hstrat_rank"]
+
+        draw_colorclade_tree(
+            frames["true_dropped_fossils"],
+            ax=axes.flat[0],
+            **plotter_kwargs,
+            label_tips=label_func,
+        )
+        draw_colorclade_tree(
+            frames["reconst_dropped_fossils"],
+            ax=axes.flat[1],
+            **plotter_kwargs,
+            label_tips=False,
+        )
+
+    draw_colorclade_tree(
+        frames["true"],
+        ax=axes.flat[-4],
+        **plotter_kwargs,
+        label_tips=label_func,
+    )
+    draw_colorclade_tree(
+        frames["reconst"],
+        ax=axes.flat[-3],
+        **plotter_kwargs,
+        label_tips=False,
+    )
+    draw_colorclade_tree(
+        true_df_no_lengths,
+        ax=axes.flat[-2],
+        **plotter_kwargs,
+        label_tips=label_func,
+    )
+    draw_colorclade_tree(
+        reconst_df_no_lengths,
+        ax=axes.flat[-1],
+        **plotter_kwargs,
+        label_tips=False,
+    )
+
+    axes.flat[0].set_xscale(
+        "function", functions=(lambda x: x**10, lambda x: x**0.1)
+    )
+    axes.flat[0].set_xlim(0, max(frames["true"]["origin_time"].unique()) + 5)
+    axes.flat[1].set_xscale(
+        "function", functions=(lambda x: x**10, lambda x: x**0.1)
+    )
+    axes.flat[1].set_xlim(
+        0, max(frames["reconst"]["origin_time"].unique()) + 5
+    )
+
+    for i in range(1, len(axes.flat), 2):
+        axes.flat[i].set_xlim(reversed(axes.flat[i].get_xlim()))
+
+    fig.set_size_inches(20, 20 + 10 * fossils)
+    if fossils:
+        axes.flat[0].set_title("True Phylogeny Dropped Fossils")
+        axes.flat[1].set_title("Reconstructed Phylogeny Dropped Fossils")
+
+    axes.flat[-4].set_title("True Phylogeny")
+    axes.flat[-3].set_title("Reconstructed Phylogeny")
+    axes.flat[-2].set_title("True Phylogeny Constant Branch Lengths")
+    axes.flat[-1].set_title("Reconstructed Phylogeny Constant Branch Lengths")
+    plt.tight_layout()
+
+
+def visualize_reconstruction(
+    frames: ReconstructionResult,
+    *,
+    visualize: bool,
```

Comment:
Rename? Having a function called `visualize_reconstruction` with a boolean flag param `visualize` makes you wonder what the flag does (or what calling with the flag False would do)